server/src/session.ts

```diff
@@ -160,6 +161,24 @@ export class Session {
     };
   }
 
+  private onGetComponentsWithTemplateFile(params: any): GetComponentsWithTemplateFileResponse
+      |undefined {
+    const lsInfo = this.getLSAndScriptInfo(params.textDocument);
+    if (lsInfo === undefined) {
+      return undefined;
+    }
+    const {languageService, scriptInfo} = lsInfo;
+    const documentSpans = languageService.getComponentLocationsForTemplate(scriptInfo.fileName);
+    const results: GetComponentsWithTemplateFileResponse = [];
+    for (const documentSpan of documentSpans) {
+      const scriptInfo = this.projectService.getScriptInfo(documentSpan.fileName);
+      const range =
+          scriptInfo ? tsTextSpanToLspRange(scriptInfo, documentSpan.textSpan) : EMPTY_RANGE;
```

Comment:
If the scriptInfo does not exist for the span then we should just skip this result.